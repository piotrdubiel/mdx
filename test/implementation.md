# Some Implementation

```ocaml
let src = Logs.Src.create "cram.test"
module Log = (val Logs.src_log src : Logs.LOG)

let (/) x y = match x with
  | "." -> y
  | _   -> Filename.concat x y

let run () _ _ _ _ _ _ _ _ _ _ =
  let base = Filename.basename Sys.argv.(0) in
  let dir = Filename.dirname Sys.argv.(0) in
  let cmd = match base with
    | "main.exe" -> dir / "test" / "main.exe"
    | x -> dir / x ^ "-test"
  in
  let argv = Array.sub Sys.argv 1 (Array.length Sys.argv - 1) in
  argv.(0) <- cmd;
  Log.debug (fun l -> l "executing %a" Fmt.(Dump.array string) argv);
  Unix.execvp cmd argv

open Cmdliner

let cmd: int Term.t * Term.info =
  let doc = "Test markdown files." in
  Term.(pure run
        $ Cli.setup $ Cli.non_deterministic $ Cli.not_verbose
        $ Cli.silent $ Cli.verbose_findlib $ Cli.prelude $ Cli.prelude_str
        $ Cli.file $ Cli.section $ Cli.root $ Cli.direction),
  Term.info "test" ~doc
```