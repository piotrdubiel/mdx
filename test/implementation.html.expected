<section id="some-implementation" class="level1">
<h1>Some Implementation</h1>
<div class="highlight">
<pre><code class="language-reason">let src = Logs.Src.create(&quot;cram.test&quot;);
module Log = (val Logs.src_log(src): Logs.LOG);

let (/) = (x, y) =&gt;
  switch (x) {
  | &quot;.&quot; =&gt; y
  | _ =&gt; Filename.concat(x, y)
  };

let run = ((), _, _, _, _, _, _, _, _, _, _) =&gt; {
  let base = Filename.basename(Sys.argv[0]);
  let dir = Filename.dirname(Sys.argv[0]);
  let cmd =
    switch (base) {
    | &quot;main.exe&quot; =&gt; dir / &quot;test&quot; / &quot;main.exe&quot;
    | x =&gt; dir / x ++ &quot;-test&quot;
    };

  let argv = Array.sub(Sys.argv, 1, Array.length(Sys.argv) - 1);
  argv[0] = cmd;
  Log.debug(l =&gt; l(&quot;executing %a&quot;, Fmt.(Dump.array(string)), argv));
  Unix.execvp(cmd, argv);
};

open Cmdliner;

let cmd: (Term.t(int), Term.info) = (
  {
    let doc = &quot;Test markdown files.&quot;;
    (
      Term.(
        pure(run)
        $ Cli.setup
        $ Cli.non_deterministic
        $ Cli.not_verbose
        $ Cli.silent
        $ Cli.verbose_findlib
        $ Cli.prelude
        $ Cli.prelude_str
        $ Cli.file
        $ Cli.section
        $ Cli.root
        $ Cli.direction
      ),
      Term.info(&quot;test&quot;, ~doc),
    );
  }: (
    Term.t(int),
    Term.info,
  )
);
</code></pre>
</div>
</section>
